packages/obonode/obojobo-chunks-list/converter.js

```diff
@@ -225,20 +229,91 @@ const switchType = {
 			})
 
-			// When the new text node is inserted, all ListLine nodes that were
-			// changed are deleted, and then the textNode is inserted at the Point
-			// before the first changed ListLine. This prevents a hanging ListLine
-			// from lingering after the ListLine nodes are changed to TextLines
-			Transforms.insertNodes(editor, textNode, {
+			// In order to properly handle the replacement without breaking normalization
+			// First delete all text that is going to be replaced
+			// This leaves us with an empty ListLine at the start path
+			// Then we delete that empty ListLine, and insert the new TextNode in its place
+			const initalPoint = Editor.start(editor, startPath)
+			const finalPoint = Editor.end(editor, endPath)
+			Transforms.delete(editor, {
 				at: {
-					anchor: Editor.before(editor, Editor.start(editor, firstPath)),
-					focus: Editor.end(editor, lastPath)
+					anchor: initalPoint,
+					focus: finalPoint
 				}
 			})
+			Transforms.removeNodes(editor, { at: startPath })
+			Transforms.insertNodes(editor, textNode, {
+				at: startPath
+			})
+
+			if (containsStart) {
+				// Since list paths have the possibility of being N deep,
+				// we use the list path to find any line children
+				// this works because there will only be one text node within a single list
+				const [firstLine] = Array.from(
+					Editor.nodes(editor, {
+						at: path,
+						match: n => n.subtype === TEXT_LINE_NODE
+					})
+				)
+
+				// The difference between startPath and start Point indicates what inline
+				// and what text leaf is currently selected
+				const leafDepth = start.path.length - startPath.length
+				const relativeLeafPath = start.path.slice(start.path.length - leafDepth)
+
+				// Therefore, the point to focus on is
+				// the line's path
+				// + the relative leaf path
+				// + the original offset
+				Transforms.setPoint(
+					editor,
+					{
+						path: firstLine[1].concat(relativeLeafPath),
+						offset: start.offset
+					},
+					{ edge: 'anchor' }
+				)
+			}
+
+			if (containsEnd) {
+				// Since list paths have the possibility of being N deep,
+				// we use the list path to find any line children
+				// this works because there will only be one text node within a single list
+				const [lastLine] = Array.from(
+					Editor.nodes(editor, {
+						at: path,
+						match: n => n.subtype === TEXT_LINE_NODE,
+						reverse: true
+					})
+				)
+
+				// The difference between endPath and end Point indicates what inline
+				// and what text leaf is currently selected
+				const leafDepth = end.path.length - endPath.length
+				const relativeLeafPath = end.path.slice(end.path.length - leafDepth)
+
+				// Therefore, the point to focus on is
+				// the line's path
+				// + the relative leaf path
+				// + the original offset
+				Transforms.setPoint(
+					editor,
+					{
+						path: lastLine[1].concat(relativeLeafPath),
+						offset: end.offset
+					},
+					{ edge: 'focus' }
+				)
+			}
```

Comment:
This code looks familiar, if multiple nodes are using it I'd like to move it into a reusable function (or if we could move some repeatable part out of it if they are indeed different)